Userland/Libraries/LibWeb/Geometry/DOMMatrixReadOnly.cpp

```diff
@@ -240,6 +241,38 @@ JS::NonnullGCPtr<DOMMatrix> DOMMatrixReadOnly::inverse() const
     return result->invert_self();
 }
 
+// https://drafts.fxtf.org/geometry/#dom-dommatrixreadonly-transformpoint
+JS::NonnullGCPtr<DOMPoint> DOMMatrixReadOnly::transform_point(DOMPointInit const& point) const
+{
+    // Let pointObject be the result of invoking create a DOMPoint from the dictionary point.
+    auto point_object = DOMPoint::from_point(realm().vm(), point);
+
+    // Return the result of invoking transform a point with a matrix, given pointObject and the current matrix. The passed argument does not get modified.
+    return transform_point(point_object);
+}
+
+// https://drafts.fxtf.org/geometry/#transform-a-point-with-a-matrix
+JS::NonnullGCPtr<DOMPoint> DOMMatrixReadOnly::transform_point(DOMPointReadOnly const& point) const
+{
+    // 1. Let x be point’s x coordinate.
+    // 2. Let y be point’s y coordinate.
+    // 3. Let z be point’s z coordinate.
+    // 4. Let w be point’s w perspective.
+    // 5. Let pointVector be a new column vector with the elements being x, y, z, and w, respectively.
+    Vector4<double> point_vector { point.x(), point.y(), point.z(), point.w() };
+
+    // 6. Set pointVector to pointVector pre-multiplied by matrix.
+    point_vector = m_matrix.transpose() * point_vector;
```

Comment:
Why are you using the transposed matrix here? It's not in the spec comments or the commit message; did we mess up with column/row ordering somewhere?

Reply:
To be honest my knowledge about linear algebra is maybe quite limited. But I thought you could reverse a vector * matrix to matrix * vector if you transposed the matrix. The API only works with matrix * vector. And this way, the result is the same as Firefox, Chrome and Safari.

Reply:
But how would you like to fix this? With the transposed matrix the results are correct, but the spec says otherwise. It could be a column/row ordering issue somewhere.

Reply:
For certain matrices you can invert the product with a vector by transposing the matrix first. The order of `vector * matrix` or `matrix * vector` is irrelevant here, since we just define the single operator with a single implementation that corresponds to a regular matrix-vector product. I.e. transposing because you switched the operands doesn't really make sense since the operation is still the same in code.

Looking at the spec, it seems to be a regular matrix-vector product:

```
Set pointVector to pointVector [pre-multiplied](https://drafts.fxtf.org/geometry/#pre-multiply) by matrix.
[...]
pre-multiply
    Term A pre-multiplied by term B is equal to B · A.
```

Could it be that DOMMatrix is using a different row/column ordering than our `Gfx::Matrix`? In that case, the constructor might be wrong.

Reply:
You are right when I rotate the values in `DOMMatrixReadOnly::initialize_from_create_2d_matrix` the correct result is calculated. I will add a commit that fixes the ordering in the constructors.